storage/stores/media_store.go

```diff
@@ -460,6 +466,112 @@ func (s *MediaStore) GetAllMediaForServer(serverName string) ([]*types.Media, er
 	return results, nil
 }
 
+func (s *MediaStore) GetUsersUsageStatsForServer(
+	serverName string,
+	orderBy string,
+	start int64,
+	limit int64,
+	fromTS int64,
+	untilTS int64,
+	searchTerm string,
+	isAscendingOrder bool,
+) ([]*types.UserUsageStats, int64, error) {
+	if !util.ArrayContains(UsersUsageStatsSorts, orderBy) {
+		return nil, 0, errors.New("invalid orderBy")
+	}
+	if start < 0 {
+		return nil, 0, errors.New("invalid start")
+	}
+	if limit < 0 {
+		return nil, 0, errors.New("invalid limit")
+	}
+
+	orderDirection := "DESC"
+	if isAscendingOrder {
+		orderDirection = "ASC"
+	}
+
+	var queryParamIdx int64 = 1
+	var commonQueryParams = []interface{}{serverName}
+
+	var filters = []string{fmt.Sprintf("origin = $%d", queryParamIdx)}
+	queryParamIdx++
+	if fromTS >= 0 {
+		filters = append(filters, fmt.Sprintf("creation_ts >= $%d", queryParamIdx))
+		queryParamIdx++
+		commonQueryParams = append(commonQueryParams, fromTS)
+	}
+	if untilTS >= 0 {
+		filters = append(filters, fmt.Sprintf("creation_ts <= $%d", queryParamIdx))
+		queryParamIdx++
+		commonQueryParams = append(commonQueryParams, untilTS)
+	}
+	if searchTerm != "" {
+		filters = append(filters, fmt.Sprintf("user_id LIKE $%d", queryParamIdx))
+		queryParamIdx++
+		commonQueryParams = append(commonQueryParams, fmt.Sprintf("@%%%s%%:%%", searchTerm))
+	}
+
+	var otherPaginationParams []interface{}
+	limitClause := fmt.Sprintf("LIMIT $%d", queryParamIdx)
+	queryParamIdx++
+	otherPaginationParams = append(otherPaginationParams, limit)
+
+	offsetClause := fmt.Sprintf("OFFSET $%d", queryParamIdx)
+	queryParamIdx++
+	otherPaginationParams = append(otherPaginationParams, start)
+
+	commonQueryPortion := fmt.Sprintf(
+		"FROM media "+
+			"WHERE %s "+
+			"GROUP BY user_id", strings.Join(filters, " AND "))
+
+	paginationQuery := fmt.Sprintf(
+		"SELECT COUNT(user_id) AS media_count, SUM(size_bytes) AS media_length, user_id "+
+			"%s "+
+			"ORDER BY %s %s "+
+			"%s "+
+			"%s;",
+		commonQueryPortion,
+		orderBy,
+		orderDirection,
+		limitClause,
+		offsetClause)
+	rows, err := s.factory.sqlDb.Query(paginationQuery, append(commonQueryParams, otherPaginationParams...)...)
```

Comment:
we should be using `QueryContext` where possible, similar to the other store functions. The context comes off the rcontext type, which should be in scope of the http request.